### PR TITLE
[MOL-14959][MA] Add banner to location-picker

### PR DESCRIPTION
**Pull request by mauw-paloit**
**Changes**

-   Add mapBannerText prop to support display of a banner in the location-picker component
- Add unit test for banner text
- Update location-field stories

**Changelog entry**

-   Add banner to `location-picker`

**Additional information**

-   You may refer to this [ticket](https://jira.ship.gov.sg/browse/MOL-14959)

### CODE REVIEW

File: src/__tests__/components/fields/location-field/location-field.spec.tsx [hunk 1]
```diff
@@ -1144,6 +1144,16 @@ describe("location-input-group", () => {
 						it.todo("should show both static map and input value and dismiss location modal");
 					});
 				});
+
+				describe("when a banner text is provided", () => {
```
Comment (ryhuz) [inline src/__tests__/components/fields/location-field/location-field.spec.tsx:1148]: can there also be a test for not showing banner text? do the components have testids appended to them?

Comment (mauw-paloit) [reply src/__tests__/components/fields/location-field/location-field.spec.tsx:1148]: Sure will add one. The location-picker component renders a `LocationPickerWrapper` which has a testid attached to it but not for the components within the wrapper.

Comment (ryhuz) [reply src/__tests__/components/fields/location-field/location-field.spec.tsx:1148]: can probably have a testid for the banner component and test for it

Comment (mauw-paloit) [reply src/__tests__/components/fields/location-field/location-field.spec.tsx:1148]: Added a testid to the banner and added a unit test

**Review by ryhuz — DISMISSED** [review]

File: src/components/fields/location-field/types.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export interface ILocationFieldSchema<V = undefined>
 		Pick<IStaticMapProps, "staticMapPinColor"> {
 	className?: string;
 	locationModalStyles?: string | undefined;
+	mapBannerText?: string;
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/components/fields/location-field/types.ts]: Updated types to add undefined

File: src/components/fields/location-field/location-modal/types.ts [hunk 3]
```diff
@@ -19,4 +19,5 @@ export interface ILocationModalProps
 	onClose: () => void;
 	onConfirm: (values: ILocationFieldValues) => void;
 	updateFormValues: (values: ILocationFieldValues, shouldDirty?: boolean) => void;
+	mapBannerText?: string;
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/components/fields/location-field/location-modal/types.ts]: Updated types to add undefined

File: src/components/fields/location-field/location-modal/location-picker/types.ts [hunk 4]
```diff
@@ -24,4 +24,5 @@ export interface ILocationPickerProps extends React.InputHTMLAttributes<HTMLDivE
 	locationAvailable: boolean;
 	gettingCurrentLocation: boolean;
 	onMapCenterChange: (latlng: ILocationCoord) => void;
+	mapBannerText?: string;
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/components/fields/location-field/location-modal/location-picker/types.ts]: Updated types to add undefined

File: src/components/fields/location-field/location-field.tsx [hunk 5]
```diff
@@ -35,6 +34,7 @@ export const LocationField = (props: IGenericFieldProps<ILocationFieldSchema>) =
 			reverseGeoCodeEndpoint,
 			staticMapPinColor,
 			validation,
+			mapBannerText,
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/components/fields/location-field/location-field.tsx]: Sorted destructured props

File: src/components/fields/location-field/location-modal/location-modal.tsx [hunk 6]
```diff
@@ -42,6 +42,7 @@ const LocationModal = ({
 	onClose,
 	onConfirm,
 	updateFormValues,
+	mapBannerText,
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/components/fields/location-field/location-modal/location-modal.tsx]: Sorted destructured props

File: src/stories/3-fields/location-field/location-field.stories.tsx [hunk 7]
```diff
@@ -148,6 +160,17 @@ Disabled.args = {
 	disabled: true,
 };
 
+export const BannerCustomisation = DefaultStoryTemplate<ILocationFieldSchema>(
+	"location-field-banner-customisation"
+).bind({});
+BannerCustomisation.args = {
+	uiType: "location-field",
+	label: "BannerCustomisation",
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/stories/3-fields/location-field/location-field.stories.tsx]: Updated label

File: src/stories/3-fields/location-field/location-field.stories.tsx [hunk 8]
```diff
@@ -40,6 +40,18 @@ const meta: Meta = {
...
+		mapBannerText: {
+			description: "Specifies the banner text to be displayed. If not specified, no banner will be displayed.",
+			table: {
+				type: {
+					summary: "string",
+				},
+			},
+			type: { name: "string", required: false },
+			control: {
+				type: "string",
```
[the remark this answers is not in the document]
Comment (mauw-paloit) [reply src/stories/3-fields/location-field/location-field.stories.tsx]: Updated

File: src/__tests__/components/fields/location-field/location-field.spec.tsx (hunk 1 above)
Comment (mauw-paloit) [reply src/__tests__/components/fields/location-field/location-field.spec.tsx]: Combined both test cases into one test suite. Thanks!

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**